src/gt4py/next/program_processors/runners/dace_fieldview/utility.py

```diff
@@ -0,0 +1,96 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2023, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from typing import Any, Mapping
+
+import dace
+
+from gt4py.next.common import Connectivity, Dimension
+from gt4py.next.iterator import ir as itir
+from gt4py.next.iterator.ir_utils import common_pattern_matcher as cpm
+from gt4py.next.program_processors.runners.dace_fieldview.gtir_python_codegen import (
+    GTIRPythonCodegen,
+)
+from gt4py.next.type_system import type_specifications as ts
+
+
+def as_dace_type(type_: ts.ScalarType) -> dace.typeclass:
+    """Converts GT4Py scalar type to corresponding DaCe type."""
+    match type_.kind:
+        case ts.ScalarKind.BOOL:
+            return dace.bool_
+        case ts.ScalarKind.INT32:
+            return dace.int32
+        case ts.ScalarKind.INT64:
+            return dace.int64
+        case ts.ScalarKind.FLOAT32:
+            return dace.float32
+        case ts.ScalarKind.FLOAT64:
+            return dace.float64
+        case _:
+            raise ValueError(f"Scalar type '{type_}' not supported.")
+
+
+def filter_connectivities(offset_provider: Mapping[str, Any]) -> dict[str, Connectivity]:
+    """
+    Filter offset providers of type `Connectivity`.
+
+    In other words, filter out the cartesian offset providers.
+    Returns a new dictionary containing only `Connectivity` values.
+    """
+    return {
+        offset: table
+        for offset, table in offset_provider.items()
+        if isinstance(table, Connectivity)
+    }
+
+
+def get_domain(
```

Comment:
Good point. However, this comment could be applied in theory to any `itir.Expr`. I understand that it makes sense here, particularly, because we are going to encounter exactly the same domain, in multiple places for one program.

Reply:
It is an interesting suggestion. Indeed, the caching could be made generic and `itir.Lambda` nodes could also be cached. I would prefer to skip it in this PR, but it could be something to consider to reduce the lowering time.

Reply:
Ok, let's skip it, but if you think the suggestion makes sense, please add a TODO to not to forget